### DISCUSSION TITLE
Concatenating two trial epochs without code duplication

### DISCUSSION THREAD

**Discussion by fspinar** [jspsych/jsPsych#3251] [Q&A]
Hello dear JsPsych community, 

I have a question regarding modularity of task / task epoch codes. Lets say I want to combine two different trial epochs within a single trial. This can be as simple as :
**fixation + choice + outcome ;** 
or more complicated :
**fixation + choice 1 + choice 2 + outcome.** 
Lets say I have my custom plugins for implementing : 
**fixation + choice 1 + outcome** and 
**fixation + choice 2 + outcome.** 
Now, in order to make the new custom plugin :
 **fixation + choice 1 + choice 2 + outcome,** 
 **_do I need to duplicate my codes from the other plugins ?_** 
 **_I was thinking if there are better ways to achieve this using jspsych functionalities._** 
 One way I can think is to implement functionalities of different trial epochs in different classes, and then use them within plugins ; which would solve the problem.
 E.g., class 1 for choice 1 ; class 2 for choice 2 ; and then use the functions of the classes within my plugins. 
 
 Is there a way to achieve something like this using jspsych without requiring to write my own classes ? 
 
 Sorry if this is confusing... 
 
 Thanks a lot !

**Comment by Shaobin-Jiang** [comment]
Why don't you use a timeline instead? Create a timeline like `[fixation, choice_timeline, outcome]`, where `choice_timeline` is a child timeline that contains your choice trials. You can then use timeline variables for that child timeline.

**Reply by fspinar** [reply]
Yes I guess I could do that. But then is the data obtained from running [fixation, choice_timeline, outcome] timeline going to be like a single-trial data ? I want to be able to have the data obtained from my trial in a single trial object, and I am not sure if this is the case when we create a timeline like this.

Thank you !!

**Reply by Shaobin-Jiang** [reply]
No. The data would be scattered across trials. But it would not be too difficult to put them together. You can add an `on_finish` event for `outcome` and add the data from previous trials to the data object collected in the current trial.
